src/core/unit-serialize.c

```diff
@@ -108,7 +108,9 @@ int unit_serialize(Unit *u, FILE *f, FDSet *fds, bool switching_root) {
         fputs(u->id, f);
         fputc('\n', f);
 
-        if (unit_can_serialize(u)) {
+        if (UNIT_VTABLE(u)->serialize) {
+                assert(UNIT_VTABLE(u)->deserialize_item);
```

Comment:
maybe move the assert up and outside of the if block, and do.

```c
assert(!UNIT_VTABLE(u)->deserialize_item == !UNIT_VTABLE(u)->serialize);
```

simply because if one is set the other should be too, and if one is unset the other shouldn't be either

Reply:
Updated like that.